src/sonic-host-services/scripts/hostcfgd

```diff
@@ -57,15 +71,188 @@ def obfuscate(data):
         return data
 
 
-def run_cmd(cmd, log_err = True):
-    try:
-        subprocess.check_call(cmd, shell = True)
-    except Exception as err:
-        if log_err:
-            syslog.syslog(syslog.LOG_ERR, "{} - failed: return code - {}, output:\n{}"
-                  .format(err.cmd, err.returncode, err.output))
-        return err.returncode
-    return 0
+class Feature(object):
+    """ Represents a feature configuration from CONFIG DB data. """
+
+    def __init__(self, feature_name, feature_cfg, device_config=None):
+        """ Initialize Feature object based on CONFIG DB data.
+
+        Args:
+            feature_name (str): Feature name string
+            feature_cfg (dict): Feature CONFIG DB configuration
+            deviec_config (dict): DEVICE_METADATA section of CONFIG DB
+        """
+
+        self.name = feature_name
+        self.state = self._get_target_state(feature_cfg.get('state'), device_config or {})
+        self.auto_restart = feature_cfg.get('auto_restart', 'disabled')
+        self.has_timer = ast.literal_eval(feature_cfg.get('has_timer', 'False'))
+        self.has_global_scope = ast.literal_eval(feature_cfg.get('has_global_scope', 'True'))
+        self.has_per_asic_scope = ast.literal_eval(feature_cfg.get('has_per_asic_scope', 'False'))
+
+    def _get_target_state(self, state_configuration, device_config):
+        """ Returns the target state for the feature by rendering the state field as J2 template. 
+        
+        Args:
+            state_configuration (str): State configuration from CONFIG DB
+            deviec_config (dict): DEVICE_METADATA section of CONFIG DB
+        Returns:
+            (str): Target feature state
+        """
+
+        if state_configuration is None:
+            return None
+
+        template = jinja2.Template(state_configuration)
+        target_state = template.render(device_config)
+        if target_state not in ('enabled', 'disabled', 'always_enabled', 'always_disabled'):
+            raise ValueError('Invalid state rendered for feature {}: {}'.format(self.name, target_state))
+        return target_state
+
+
+class FeatureHandler(object):
+    """ Handles FEATURE table updates. """
+
+    SYSTEMD_SYSTEM_DIR = '/etc/systemd/system/'
+    SYSTEMD_SERVICE_CONF_D = os.path.join(SYSTEMD_SYSTEM_DIR, '{}.service.d/')
```

Comment:
Do we need rename the variable `SYSTEMD_SERVICE_CONF_D` to `SYSTEMD_SERVICE_CONF_DIR`?

Reply:
Ok